src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,173 @@
+
+interface CalendarProps {
+    initialEvents: CalendarEvent[];
+    view?: string;
+    editable?: boolean;
+    handleDateClick?: (info: DateClickArg) => void;
+}
+
+export interface CalendarEvent {
+    id: string;
+    title: string;
+    start: Date;
+    end: Date;
+    description?: string;
+    daysOfWeek?: number[];
+    startRecur?: string;
+    endRecur?: string;
+    backgroundColor?: string;
+    borderColor?: string;
+    textColor?: string;
+    allDay?: boolean;
+}
+
+const StyledDialog = styled(Dialog)`
+    // root
+    & > div {
+        // container
+        > div {
+            //Paper
+            width: min(50%, 600px);
+            border-radius: 2rem;
+            padding: 1.5rem;
+            text-align: center;
+        }
+    }
+`;
+
+const StyledCancelButton = styled.button`
+    position: absolute;
+    right: 20px;
+    top: 20px;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    row-gap: 1rem;
+`;
+
+const CalendarStyling = styled.div`
+    .fc {
+        background-color: white;
+        border-radius: 2rem;
+        padding: 1.5rem;
+        text-align: center;
+    }
+
+    .fc .fc-toolbar-title {
+        font-size: 1.5rem;
+    }
+
+    .fc .fc-toolbar-chunk {
+        margin: 0 0.5rem;
+    }
+
+    .fc .fc-button {
+        height: max(3vmin, 30px);
+        font-size: max(1.5vmin, 15px);
+        text-align: center;
+        vertical-align: middle;
+        padding: 0 10px 0 10px;
+    }
+
+    .fc .fc-icon {
+        font-size: max(1.5vmin, 15px);
+    }
+
+    .fc .fc-button-primary {
+        background-color: ${(props) => props.theme.fillColor};
+        border-color: ${(props) => props.theme.fillColor};
+        color: black;
+    }
+
+    .fc .fc-button-primary:hover {
+        background-color: ${(props) => props.theme.fillColor};
+        border-color: ${(props) => props.theme.fillColor};
+        color: black;
+    }
+
+    .fc .fc-button-primary:not(:disabled).fc-button-active,
+    .fc .fc-button-primary:not(:disabled):active {
+        background-color: ${(props) => props.theme.fillColor};
+        border-color: ${(props) => props.theme.fillColor};
+        color: black;
+    }
+`;
+
+const Calendar: React.FC<CalendarProps> = ({
+    initialEvents,
+    view = "dayGridMonth",
+    editable = false,
+    handleDateClick,
+}) => {
+    const [eventClick, setEventClick] = useState<CalendarEvent | null>(null);
+
+    function handleEventClick(info: EventClickArg): void {
+        const id = info.event.id;
+        setEventClick(initialEvents.find((e) => e.id === id) ?? null);
+    }
+
+    const dateFormatOptions: Intl.DateTimeFormatOptions = {
+        weekday: "long",
+        year: "numeric",
+        month: "long",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+    };
+
+    return (
+        <>
+            <StyledDialog open={eventClick !== null} onClose={() => setEventClick(null)}>
```

Comment:
can we split the `StyledDialog` into a separate file? this is quite long and seems like two separate parts - the 'Calendar' and the 'Modal that opens', so should be in two separate files (but can both be in `components/Calendar`!)
This may also be the file that is unified later!

Reply:
it doesn't have to be a split with a lot of thought, especially since this is likely to be rewritten soon, but I think it's likely we'll use this code as a reference (so should be 'pretty clean' anyway!)